backend/routes/user-route.js

```diff
@@ -681,6 +686,205 @@ router.get('/offered-bookings', authenticateJWT, async (req, res) => {
+            }
+        });
+
+        if (!booking) {
+            return res.status(404).json({ error: 'Booking not found' });
+        }
+
+        // Ensure the booking belongs to the authenticated user or the service provider is the authenticated user
+        if (booking.userId !== req.user.id && booking.service.userId !== req.user.id) {
+            return res.status(403).json({ error: 'You are not authorized to cancel this booking' });
+        }
+
+        // Delete the booking
+        await prisma.booking.delete({
+            where: { id: bookingId }
+        });
+
+        // Update the time slot to mark it as not booked
+        await prisma.availableTime.update({
+            where: { id: booking.timeId },
+            data: { isBooked: false }
+        });
+
+        const notificationContentForUser = `You canceled your booking of ${booking.service.serviceName} with ${booking.service.user.username} for ${new Date(booking.time.startTime).toLocaleString()}`;
+        const notificationContentForServiceProvider = `${req.user.username} canceled their booking of ${booking.service.serviceName} with you for ${new Date(booking.time.startTime).toLocaleString()}`;
+
+        // Create notification for the user who canceled the booking
+        await prisma.notification.create({
+            data: {
+                content: notificationContentForUser,
+                userId: booking.user.id,
+                serviceId: booking.service.id
+            }
+        });
+
+        // Increment unread count for the user who canceled the booking
+        await prisma.user.update({
+            where: { id: booking.user.id },
+            data: {
+                unreadCount: {
+                    increment: 1
+                }
+            }
+        });
+
+        // Create notification for the service provider
+        await prisma.notification.create({
+            data: {
+                content: notificationContentForServiceProvider,
+                userId: booking.service.user.id,
+                serviceId: booking.service.id
+            }
+        });
+
+        // Increment unread count for the service provider
+        await prisma.user.update({
+            where: { id: booking.service.user.id },
+            data: {
+                unreadCount: {
+                    increment: 1
+                }
+            }
+        });
+
+        const cancellationDetails = {
+            userId: req.user.id,
+            serviceId: booking.service.id,
+            timeId: booking.time.id,
+            userName: req.user.username,
+            serviceName: booking.service.serviceName,
+            businessName: booking.service.user.username,
+            cancellationTime: new Date().toISOString(),
+            serviceCreatorId: booking.service.userId
+        };
+
+        sendNotification(cancellationDetails);
+
+        res.status(200).json({ message: 'Booking canceled successfully' });
```

Comment:
One small idea that would help with readability. Convert these to variable to match the code. Codes are here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses

Example:
```
const OK = 200
```

then in you can do `res.status(OK)`